r2dbc-spi/src/main/java/io/r2dbc/spi/Result.java

```diff
@@ -29,13 +32,56 @@
  */
 public interface Result {
 
+    /**
+     * A mapping function container that is used to map interleaved result types of unknown order.
+     *
+     * <p>Databases may produce interleaved update counts or result sets without explicit knowledge of order by clients.</p>
+     */
+    interface Mapping<T> {
```

Comment:
I don't have yet a good idea of how useful `T` really is when working with combined input types (beyond using containers such as `Optional<Object>` or `AtomicReference<Object>`).

Reply:
If there's no `<T>` on this wrapper type, then the `map(Mapping<T>)` method cannot capture the type, and is forced to return `Publisher<?>`.

When a stream of mixed content is produced and processed, there is a choice of:

- Wrapping everything in a new content agnostic wrapper, just like `Result`
- Working with `Object`

`<T>` may not always be useful, in case of which, it can be bound to `Object`. But when the user wants their custom wrapper, then they can't get type safety without this `<T>`.